pkg/controllers/vrpq/query_reconciler.go

```diff
@@ -71,66 +79,108 @@ func (q *QueryReconciler) Reconcile(ctx context.Context, req *ctrl.Request) (ctr
 		return res, err
 	}
 
-	return ctrl.Result{}, q.runListRestorePoints(ctx, req)
+	finder := iter.MakeSubclusterFinder(q.VRec.Client, q.Vdb)
+	pods, err := finder.FindPods(ctx, iter.FindExisting)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// find a pod to execute the vclusterops API
+	var podIP string
+	for i := range pods.Items {
+		pod := &pods.Items[i]
+		if pod.Status.Phase == podRunning {
+			for j := range pod.Status.ContainerStatuses {
+				// check NMA container ready
+				if pod.Status.ContainerStatuses[j].Ready {
```

Comment:
You need to check the container name first to make sure it is indeed the `nma`(pod.Status.ContainerStatuses[j].Name == names.NMAContainer`.

Reply:
Yes, I realized that too. Actually, I have a question: Do we only have an NMA container if we define the annotation vertica.com/run-nma-in-sidecar: "true"? What if we don't define this annotation and we only have a server container? If we include the condition pod.Status.ContainerStatuses[j].Name == names.NMAContainer, we will not find any pods that satisfy the above conditions.